batch_job/src/batch_job_local.c

```diff
@@ -110,20 +110,23 @@ static batch_job_id_t batch_job_local_wait (struct batch_queue * q, struct batch
 
 static int batch_job_local_remove (struct batch_queue *q, batch_job_id_t jobid)
 {
-	if(kill(jobid, SIGTERM) == 0) {
+	while(kill(jobid, SIGTERM) == 0) {
```

Comment:
We may want to try only a handful of times, and then send a SIGKILL. If the batchjob ignores the SIGTERM, then we wait forever.

Reply:
I wonder if it would be too difficult to add a function that gets us the pid's of all the children of jobid, like `pgrep` does, and repeat here what you do for poncho, that is, send a kill to all the children too.

Reply:
I added a functionality so that it attempts up to 12 times (60 seconds) for a clean termination with SIGTERM, and then tries another 12 times/60 seconds sending SIGKILL, returning 0 if this is still not successful.
Doesn't work_queue_factory already track all of the child processes/jobs and when it calls remove_all_workers it calls batch_job_local_remove on each of the children, so wouldn't have a function to do the same thing be repetitive?